packages/runtime-core/src/vnode.ts

```diff
@@ -286,12 +286,12 @@ function setupBlock(vnode: VNode) {
  * @private
  */
 export function createElementBlock(
-  type: string,
-  props?: Record<string, any> | null,
-  children?: any,
+  type: VNodeTypes | ClassComponent | typeof NULL_DYNAMIC_COMPONENT,
```

Comment:
Strictly speaking, the `type` cannot be a component type.

Reply:
Why is that and why isn't that reflected in `createBaseVNode`?

Reply:
`createElementBlock` will only be generated in callsites where the argument is either a plain element or a fragment. This is defined by how it is generated by the compiler.